### DISCUSSION TITLE
What is minimum version of React required for react-flow...

### DISCUSSION THREAD

**Discussion by horstheistermann** [xyflow/xyflow#1734] [General]
HI I would like to use this library but I need to run it in an environment that is running react ^16.5.2
Is there a version of react-flow that can run inside a react app this old?

**Comment by leejustin** [comment]
Hi @horstheistermann You can do this yourself by checking the required versions of `@react` and how they change in the commits of `package.json`. According to the [commit history](https://github.com/wbkd/react-flow/commit/f244e77b9eac16433415a33d0c39f68cd5b56fc2#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519), at _least_ `9.5.4` supports 16. You can look through the commit history to find the latest version prior to when support for 16 is dropped.

**Reply by horstheistermann** [reply]
thanks for the info!

**Reply by moklick** [reply]
We should still support React v16. Aren't we? https://github.com/wbkd/react-flow/blob/main/package.json#L78

**Comment by horstheistermann** [comment]
yes it does work with react 16  thanks!
